core/kamon-core/src/main/scala/kamon/ContextStorage.scala

```diff
@@ -141,8 +141,10 @@ object ContextStorage {
     * instrumentation follows them around.
     */
   private val _contextStorage: Storage = {
-    if(sys.props("kamon.context.debug") == "true")
+    if (sys.props("kamon.context.debug") == "true")
```

Comment:
Now that the choice is between a couple of actual implementations, this should probably be loaded from the configuration file, not the environment

Reply:
Folks, I'm thinking that we should make `CrossThreadLocal` the default context storage implementation. The reasons why I think that are:
- It will not affect the behavior of any of the manual/automatic instrumentation. We always aimed to close scopes on the same thread where we created them, and in those situations this implementation will produce the same results. In 99% of the cases we need to close the scope in the same thread we created it anyways, and Cats/Monix implementations are an exception to that common case so we will continue to recommended closing scopes on the same thread, unless it is very clear that there is a reason for doing otherwise, as it is for Cats/Monix.
- It makes it easier to get started. If there is something I learned over the years is that users almost never read the fine print :smile:.. if users need to add a JVM property but that property is controlled from the outside world (IDE, command line parameters or container orchestrators), there will be several cases of things "not working" in certain environments. I would rather go with an option that makes it work by default for everyone.
- In the specific cases where someone needs that little boost in performance (I'm counting to be very few cases here) they can add the extra JVM options to use the optimized TLS.

What do you think about that?

Reply:
This all sounds reasonable, and simplifying semantics is always good!
I'm just scared because this is a "breaking" change, and I have no intuition for how often this code is called. 
If you're sure that we won't see a big performance drop, let's do it, and just make sure to document clearly and make not of it in the release notes.

Reply:
I totally agree @ivantopo (Makes my life easier that's for sure ;)). Happy to make the change if @SimunKaracic agree too

Reply:
I agree :handshake:

Reply:
This is done.